homeassistant/bootstrap.py

```diff
@@ -325,15 +325,29 @@ def _get_domains(hass: core.HomeAssistant, config: Dict[str, Any]) -> Set[str]:
     hass: core.HomeAssistant, config: Dict[str, Any]
 ) -> None:
     """Set up all the integrations."""
+
+    async def async_setup_multi_components(domains: Set[str]) -> None:
+        """Set up multiple domains. Log on failure."""
+        futures = {
+            domain: hass.async_create_task(async_setup_component(hass, domain, config))
+            for domain in domains
+        }
+        await asyncio.wait(futures.values())
+        errors = [domain for domain in domains if futures[domain].exception()]
+        for domain in errors:
+            _LOGGER.error(
```

Comment:
For this one, we have set up `async_setup_component` in such a way that we already wrap every piece of code that we call from an integration with `try…catch`. So if any of this raises, it's truly unexpected. So we should make sure we pass the exception as `exc_info` so we can print a stack trace

Reply:
ok. done